File: sdks/java/lib/src/main/java/models/shared/RetentionPolicyAssignmentBase.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * RetentionPolicyAssignmentBase
 * A base representation of a retention policy assignment.
**/
public class RetentionPolicyAssignmentBase {
    @JsonProperty("id")
    public String id;
    public RetentionPolicyAssignmentBase withId(String id) {
        this.id = id;
        return this;
    }
    @JsonProperty("type")
    public RetentionPolicyAssignmentBaseTypeEnum type;
    public RetentionPolicyAssignmentBase withType(RetentionPolicyAssignmentBaseTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/retentionpolicyassignmentbase.go
package shared

type RetentionPolicyAssignmentBaseTypeEnum string

const (
	RetentionPolicyAssignmentBaseTypeEnumRetentionPolicyAssignment RetentionPolicyAssignmentBaseTypeEnum = "retention_policy_assignment"
)

// RetentionPolicyAssignmentBase
// A base representation of a retention policy assignment.
type RetentionPolicyAssignmentBase struct {
	ID   string                                `json:"id"`
	Type RetentionPolicyAssignmentBaseTypeEnum `json:"type"`
}
